.github/workflows/ci.yml

```diff
@@ -292,13 +313,40 @@ jobs:
         with:
           timeout_minutes: 60
           max_attempts: 2
-          command:  bundle exec rake test:multiverse[group=${{ matrix.multiverse }},verbose]
+          command: bundle exec rake test:multiverse[group=${{ matrix.multiverse }},verbose]
         env:
           DB_PORT: ${{ job.services.mysql.ports[3306] }}
           SERIALIZE: 1
           JRUBY_OPTS: --dev
 
-
       - name: Annotate errors
         if: ${{ failure() }}
         uses: ./.github/actions/annotate
+
+  simplecov:
+    needs: [unit_tests, multiverse, infinite_tracing]
+    runs-on: ubuntu-20.04
+    if: ${{ always() }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
+      - run: bundle
+      - name: Download all workflow run artifacts
+        uses: actions/download-artifact@v3
+      - name: Collate Coverage Results
+        run: bundle exec rake coverage:report
+      - name: Upload coverage results
+        uses: actions/upload-artifact@master
+        with:
+          name: coverage-report-combined
+          path: lib/coverage
+      - name: Simplecov Report
+        uses: ./.github/actions/simplecov-report
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          resultPath: lib/coverage/.last_run.json
+          failedThreshold: 60
+          failedThresholdBranch: 30
+        
```

Comment:
@hannahramadan it looks like this file lost its trailing newline and should have it restored. These instructions for having VS Code do it automatically look right from what I can recall with that editor:

https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

Reply:
Unfortunately I'm still on homework assignment: automatically add `end` 🥲 But I was able to get this final newline added! Thanks for the catch and resource for setting up this check.